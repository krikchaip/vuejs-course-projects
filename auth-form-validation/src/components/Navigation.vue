<template>
  <div class="navigation">
    <router-link class="navigation__brand" to="/">
      Vue - Complete Guide
    </router-link>
    <div class="navigation__group">
      <slot></slot>
    </div>
  </div>
</template>
