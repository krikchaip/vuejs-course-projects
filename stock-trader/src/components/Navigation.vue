<template>
  <nav class="navbar navbar-expand navbar-light bg-light shadow-sm">
    <router-link
      class="navbar-brand"
      to="/">
      Stock Trader
    </router-link>
    <div class="navbar-nav mr-auto">
      <router-link
        active-class="active"
        class="nav-item nav-link"
        to="/portfolio">
        Portfolio
      </router-link>
      <router-link
        active-class="active"
        class="nav-item nav-link"
        to="/stocks">
        Stocks
      </router-link>
    </div>
    <div class="navbar-nav">
      <strong class="nav-item navbar-text">
        Funds: ${{ funds.toLocaleString() }}
      </strong>
    </div>
  </nav>
</template>

<script>
  export default {
    props: {
      funds: { type: Number, default: 0 }
    }
  }
</script>

<style scoped>
  .nav-item {
    font-size: 0.95rem;
  }
</style>
