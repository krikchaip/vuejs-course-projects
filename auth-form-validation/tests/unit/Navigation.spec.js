import { shallowMount } from '@vue/test-utils'
import Navigation from '@/components/Navigation'

describe('default slot', () => {
  describe('links', () => {
    let wrapper
    beforeEach(() => {
      wrapper = shallowMount(Navigation, {
        slots: {
          default: ['<link/>', '<link/>', '<link/>']
        }
      })
    })

    it('should render inside their group', () => {
      const group = wrapper.find('.navigation__group')
      expect(group.html()).toMatchSnapshot()
    })
  })
})
