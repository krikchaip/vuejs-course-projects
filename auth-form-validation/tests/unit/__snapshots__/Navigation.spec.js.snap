// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default slot links should render inside their group 1`] = `
<div class="navigation__group">
  <link>
  <link>
  <link>
</div>
`;
